src/rgw/driver/sfs/sqlite/dbconn.cc

```diff
@@ -187,6 +196,24 @@ StorageRef DBConn::get_storage() {
   }
 }
 
+dbapi::sqlite::database DBConn::get() {
```

Comment:
If a thread already called get_storage() this returns that connection (set in on_open) with the sqlite init we exect (busy timeout, error log, profiling settings)

If that thread did not, this is a new connection with sqlite modern cpp default sqlite_open settings. 

We should avoid mixing sqlite connections with different initializations. 

Would it suffice to just call get_storage() in the catch, ignore the result and take storage_pool_new.at() which was set in on_open?

Reply:
Good point (I was wondering about that too). I suspect that workaround you suggested should work.

Reply:
Yup, that's something I was afraid of. I will try that workaround.